### PR TITLE
Add an action to add trunk to live branches manifest

### PR DESCRIPTION
**Pull request by chihsuan**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

p1684816055228779-slack-C03CPM3UXDJ

This PR adds a Github action to add trunk to the branches manifests so that our stakeholder can view the up-to-date progress of our works.

It should do the same thing as [`pr-build-live-branch.yml`](https://github.com/woocommerce/woocommerce/blob/cced23b2db5ac387ceb71b29999202c5831a54bc/.github/workflows/pr-build-live-branch.yml). I copied it from that file and only made changes to the trigger rules and removed the ["PR" parameter](https://github.com/woocommerce/woocommerce/blob/cced23b2db5ac387ceb71b29999202c5831a54bc/.github/workflows/pr-build-live-branch.yml#L86) from request.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

- Review the changes
- I'm not sure how to test it. Perhaps, check out a new branch and add it to "push branches" to test the action.

### CODE REVIEW

**Comment by adrianduffell** [comment]
Hmm yes this seems very tricky to test indeed! 🤔 

@ObliviousHarmony @samueljseay do you have any ideas how to test it?

**Comment by ObliviousHarmony** [comment]
Honestly, with something like this, the only way to test it is to merge it :smile:

File: .github/workflows/build-live-branch.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                  unzip woocommerce.zip
+                  rm woocommerce.zip
+                  mv woocommerce woocommerce-dev
+                  zip -q -r "woocommerce-dev.zip" "woocommerce-dev/"
+                  rm -fR "$GITHUB_WORKSPACE/zips/woocommerce-dev"
+
+                  # Plugin data is passed as a JSON object.
+                  PLUGIN_DATA="{}"          
+                  PLUGIN_DATA=$( jq -c --arg slug "woocommerce" --arg ver "$CURRENT_VERSION" '.[ $slug ] = { version: $ver }' <<<"$PLUGIN_DATA" )
+                  echo "plugin-data=$PLUGIN_DATA" >> $GITHUB_OUTPUT
```
Comment (adrianduffell) [inline .github/workflows/build-live-branch.yml:66]: @ObliviousHarmony , as a follow-up task, would it be worth moving the run commands to a dedicated script to be shared with the PR action? Just seems we might create a maintenance headache here with similar code in 2 places.

**Review by adrianduffell — APPROVED** [review]
LGTM thank you @chihsuan! I'm willing to test it by merging if @ObliviousHarmony is!

File: .github/workflows/build-live-branch.yml (hunk 1 above)
Comment (ObliviousHarmony) [reply .github/workflows/build-live-branch.yml:66]: I've got an agenda item for our team meetup at the start of June to go through all of our workflows and basically rebuild them. We have a _lot_ of redundant workflows that can be cleaned up, inconsistent job ordering, and all sorts of other things that need to be worked on. We will take a look at this one at that :smile:

**Pull request merged**